checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -28,33 +36,73 @@
  *   <li value="2">The passed array size is less than the collection size. Here are heuristics to
  *       handle the most common cases:
  *       <ol>
- *         <li value="1">an empty array initializer, e.g. {@code c.toArray(new String[] {})},
- *         <li value="2">array creation tree of size 0, e.g. {@code c.toArray(new String[0])}, or
- *         <li value="3">array creation tree of the collection size method invocation {@code
- *             c.toArray(new String[c.size()])}
+ *         <li value="1">the argument has length 0:
+ *             <ol>
+ *               <li value="1">an empty array initializer, e.g. {@code c.toArray(new String[] {})},
+ *                   or
+ *               <li value="2">array creation tree of size 0, e.g. {@code c.toArray(new String[0])}.
+ *             </ol>
+ *         <li value="2">array creation tree with a collection {@code size()} method invocation as
```

Comment:
The change in numbering makes the comments in `isHandledArrayCreation` inaccurate.  Please update them along with this update.  And add numbering for the code that calls `isArrayLenZeroFieldAccess`, though folding it into `isHandledArrayCreation` is probably a better idea, see below.